### DISCUSSION TITLE
Query result type in docs does not match Executed type in code

### DISCUSSION THREAD

**Discussion by danboyle8637** [planetscale/discussion#543] [FAQs]
According to the serverless docs here: https://planetscale.com/docs/tutorials/planetscale-serverless-driver

The query result should return...
```
{
  headers: [],
  types: {},
  rows: [],
  rowsAffected: 1,
  insertId: '6',
  error: null,
  size: 0,
  statement: "INSERT INTO hotels (`name`, `address`, `stars`) VALUES ('Montage Kapalua Bay 2', '1 Bay Dr, Lahaina, HI 96761', 4)",
  time: 102
}
```

In my code when I look at the `ExecutedQuery` type... at least I think I remember that correctly...

The `error` property is not available.

Is there a better way to catch query errors if we don't have access to the `error` property... or is that not even what this property is used for?

Thanks for any help

**Comment by orware** [comment]
Hello @danboyle8637,

If you are able to provide an example of the response you're receiving that would be helpful to compare against an example I'm looking at pulled from a test via the web console on my end, since our web console also uses the serverless driver.

However, based on what you've shared I am wondering if changes like the one linked to below might provide an explanation for the difference in output at least for the `error` property:
https://github.com/planetscale/database-js/issues/61

The related pull request seems to indicate that the `error` property went away as a part of the merged changes:
https://github.com/planetscale/database-js/pull/62/files

In other words, the documentation in the example for the page you linked to is likely outdated due to changes that have occurred in the serverless driver since that page was first written and we may need to review that page for accuracy if the output of the serverless driver is different now.

**Reply by danboyle8637** [reply]
@orware Sorry for the late reply. 

Here's the definition of an `ExecutedQuery` from the type definition file: 

```
export interface ExecutedQuery {
    headers: string[];
    types: Types;
    rows: Row[];
    fields: Field[];
    size: number;
    statement: string;
    insertId: string;
    rowsAffected: number;
    time: number;
}
```

The `error` property is not there. 

And when I'm doing say an UPDATE query... what is the best way to handle an error. I'm guessing the query will error throw an error automatically that my catch block will catch?

I know I can test this but I would like to konw if there is a "best practices" for handling errors from queries.

Thanks!

**Reply by orware** [reply]
No problem @danboyle8637,

Yup, that lines up with the pull request I linked to above and the related changes. So rather than that `error` property being present, if the `await` / Promise triggers an error it is instead caught within a `catch` block after those changes occurred.

You can see this and likely a few other examples / responses present within the serverless driver's tests here which probably will do a better job of showing current examples as well:
https://github.com/planetscale/database-js/blob/main/__tests__/index.test.ts#L404-L407

I hope this helps make things clearer!

**Comment by danboyle8637** [comment]
Awesome thanks @orware !!
